animation_nodes/data_structures/meshes/mesh_data.pyx

```diff
@@ -140,37 +150,51 @@ cdef class Mesh:
         if self.getAttributeDomainLength(attribute.domain) != len(attribute.data):
             raise Exception("invalid length")
 
-        self.attributes[attribute.name] = attribute
+        if attribute.type == AttributeType.MATERIAL_INDEX:
+            self.materialIndices = attribute
+        elif attribute.type == AttributeType.UV_MAP:
+            self.uvMapAttributes[attribute.name] = attribute
+        elif attribute.type == AttributeType.VERTEX_COLOR:
+            self.vertexColorAttributes[attribute.name] = attribute
+        elif attribute.type == AttributeType.MATERIAL_INDEX:
+            self.materialIndices = attribute
+        else:
+            self.customAttributes[attribute.name] = attribute
 
     def getAttributes(self, AttributeType type):
```

Comment:
I would specialize all of those method. `getAllCustomAttributes`, `getAllVertexColors`, `getAllUVMaps`.

Also, you don't need to return the name, the name is already a member of the attribute.